### PR TITLE
Show dispute details on the transaction details screen when a dispute is awaiting response and past due

### LINKED ISSUE #7229: Transaction Details → Dispute Details → show details when dispute is awaiting response yet past due
Part of #6883

We should show the details of a dispute on the Transaction Details screen when a dispute is in a `needs_response`/`warning_needs_response` status, and the `due_by` date is in the past.

We currently hide all dispute details in this case – it would be more robust to handle this gracefully, however unlikely the case may be.

This case does occur when in test mode since disputes created in test mode remain as `needs_reponse`/`warning_needs_response` status until explicit action is taken (challenge or accept), even if it is past the `due_by` date.

**Current Transaction Details screen** when a dispute is awaiting a response, `due_by` in the past:
image

**Suggested Transaction Details screen** when a dispute is awaiting a response, `due_by` in the past:
image

Note the **(Past due)** text.

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #7229

> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`

#### Changes proposed in this Pull Request

This PR allows the Transaction Details dispute details to be rendered if a dispute is awaiting a response but is past its `due_by` date.

It is uncertain whether this could occur for accounts in live mode since a dispute will move to a `won`/`lost`/`under_review` status once the `due_by` date has passed.

This case does occur when in test mode since disputes created in test mode remain as `needs_reponse`/`warning_needs_response` status until explicit action is taken (challenge or accept), even if it is past the `due_by` date.

#### Screenshots

**Current Transaction Details screen** when a dispute is awaiting a response, `due_by` in the past:
image

**Suggested Transaction Details screen** when a dispute is awaiting a response, `due_by` in the past:
image

Note the **(Past due)** text.

#### Testing instructions

* Enable feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/) or by modifying your database table `wp_options` directly.
* Create a disputed transaction using the card `4000000000000259` at checkout.
* Wait 7+ days ⚠️ 
* Find the dispute in Payments → Disputes and click it to view the dispute details screen.
* Click on the Transaction ID to view the Transaction Details screen.
* On `trunk`, you will not see any dispute details, even though the dispute's status is `Needs response`.
* On this PR, you should see the dispute details.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/payment-details/dispute-details/dispute-awaiting-response-details.tsx [hunk 1]
```diff
@@ -34,33 +33,25 @@ const DisputeAwaitingResponseDetails: React.FC< Props > = ( { dispute } ) => {
 		<div className="transaction-details-dispute-details-wrapper">
 			<Card>
 				<CardBody className="transaction-details-dispute-details-body">
-					{ isAwaitingResponse( dispute.status ) &&
```
Comment (Jinksi) [inline client/payment-details/dispute-details/dispute-awaiting-response-details.tsx:62]: Note: `isAwaitingResponse` is handled in the parent component:

https://github.com/Automattic/woocommerce-payments/blob/050ee037ba7575f04477d6451ee16400823720ef/client/payment-details/summary/index.tsx#L381-L391

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7231` or branch name `fix/7228-dispute-details-when-awaiting-response-past-due` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7228-dispute-details-when-awaiting-response-past-due) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 0080da1b25e0009cfd29a272d75219da7e710ec7
- Build time: 2023-09-21 21:34:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by shendy-a8c — COMMENTED** [review]
Since https://github.com/Automattic/woocommerce-payments/pull/7206/files#r1325199048 copied `DisputeSummaryRow`'s 'smart day' and this PR changes that behavior, I think the change (showing `Past due`) should also be applied to `DisputeSteps` (awaiting for https://github.com/Automattic/woocommerce-payments/pull/7206 to be merged).

Other than that, this PR looks good and useful. I have been seeing the empty rectangle for the past due disputes on my local test store.

**Comment by Jinksi** [comment]
> Since https://github.com/Automattic/woocommerce-payments/pull/7206/files#r1325199048 copied DisputeSummaryRow's 'smart day' and this PR changes that behavior, I think the change (showing Past due) should also be applied to DisputeSteps (awaiting for https://github.com/Automattic/woocommerce-payments/pull/7206 to be merged).

We can address this in a subsequent issue/PR to allow us to test this PR in the release testing today.

New issue created for this: #7255

**Review by shendy-a8c — APPROVED** [review]

**Pull request merged**